tests/test_other.py

```diff
@@ -136,6 +138,20 @@ def do_other_test(self, dirname, emcc_args=[], run_args=[]):
     seen = run_js('a.out.js', args=run_args, stderr=PIPE, full_output=True) + '\n'
     self.assertContained(expected, seen)
 
+  def do_smart_test(self, source, name='test.cpp', literals=[], regexes=[],
+                    emcc_args=[], run_args=[], assert_returncode=0):
+    shutil.copyfile(source, name)
+    run_process([PYTHON, EMCC, name] + emcc_args)
```

Comment:
`self.get_emcc_args()` is better I think

Reply:
I don't think anything in `test_other.py` uses `self.emcc_args`.

Reply:
I recall that @kripken wants no tests in `test_other` to use `self.emcc_args` and the mechanism behind that.

Reply:
I believe that is incorrect.  For `self.set_setting()` and `self.emcc_args` shouldn't work for all test subclasses.   Thats the point of the common code in runner.py.  @kripken can you confirm?

The `set_setting`/`get_setting`/`emcc_args`/`build` stuff in RunnerCore is all generic right?